src/ledger/LedgerManagerImpl.cpp

```diff
@@ -530,34 +538,34 @@ LedgerManagerImpl::historyCaughtup(asio::error_code const& ec,
     }
     else
     {
-        // If we're in CATCHUP_RECENT mode, we actually only got part way
-        // through catchup -- we did the minimal prefix part -- and will
-        // get another callback as CATCHUP_COMPLETE when recent-replay is
-        // done. So for now just deposit LCL and prepare for replay.
-        if (mode == CatchupManager::CATCHUP_RECENT)
+        switch (progressState)
         {
-            mLastClosedLedger = lastClosed;
-            CLOG(INFO, "Ledger") << "First phase of CATCHUP_RECENT done: "
-                                 << ledgerAbbrev(mLastClosedLedger);
-            mCurrentLedger = make_shared<LedgerHeaderFrame>(lastClosed);
-            return;
-        }
+            case CatchupWork::ProgressState::APPLIED_BUCKETS:
+            {
+                mCurrentLedger = make_shared<LedgerHeaderFrame>(lastClosed.header);
+                storeCurrentLedger();
```

Comment:
I didn't realize that we were not saving the state during catching up (internally to `CatchupWork`)... isn't that a potential source of problem if we only do partial catchup? Like if we apply buckets and crash/timeout when downloading transactions?

Reply:
this question still stands

Reply:
In previous version we were not storing state on bucket apply, because we were always applying buckets after that. Even in CATCHUP_MINIMAL case it was replaying buffered transactions (we always had at least one).

Now with commandline catchup case buffered transactions are always empty, so we need to store some state in database.

And that case is similar to applying buckets and then crashing during downloading transactions. Database will be in correct state.

Reply:
I see, thanks for the clarification & background on why this is needed. Maybe @jonjove will want to move this closer to the actual "apply buckets" code in his PR (and if invariants are executed in command line mode)